docs/jordan_gradient.rst

```diff
@@ -0,0 +1,16 @@
+Jordan Gradient
+==========================
+
+Overview
+--------
+
+This is an implementation of Stephen Jordan's "Fast Quantum Algorithm for
+Numerical Gradient Estimation" [1]_. It utilizes an oracle to sample some
```

Comment:
Actually similar to below, this is not how it's done in the cited paper, so somewhere we should make clear that the current implementation is not identical to the cited paper.

In particular, a discussion about how the implemented version is more expensive might be good to do. If I recall correctly, we are both defgate-ing a unitary (which is expensive to compile) and doing phase estimation, which has the QFT as a subroutine.

Reply:
This implementation isn't 100% true to the paper but I think it may be closer than you think.

Jordan does:
1. Hadamard -> IQFT to initilize
2. "Next, use the blackbox to compute f and add it modulo N_o into the output register. The output register is an
eigenstate of addition modulo N_o. The eigenvalue corresponding to addition of x is e^{i2πx/N_o}. Thus by writing into the output register via modular addition, we obtain a phase proportional to f. This technique is sometimes called phase kickback."
3. QFT

The way phase estimation in grove is essentially a combination of 2 and 3. So it's actually quite close to Jordan's algorithm.

That being said, I can make this more clear!